src/Serialization/Rings.jl

```diff
@@ -60,35 +62,40 @@ end
 @registerSerializationType(zzModMPolyRing)
 @registerSerializationType(zzModPolyRing)
 
-function save_internal(s::SerializerState, R::Union{MPolyRing, PolyRing})
+function save_internal(s::SerializerState, R::Union{UniversalPolyRing, MPolyRing, PolyRing})
     return Dict(
         :symbols => save_type_dispatch(s, symbols(R)),
         :base_ring => save_type_dispatch(s, base_ring(R)),
     )
 end
 
 function load_internal(s::DeserializerState,
-                       T::Type{<: Union{MPolyRing, PolyRing}},
+                       T::Type{<: Union{UniversalPolyRing, MPolyRing, PolyRing}},
                        dict::Dict)
     base_ring = load_unknown_type(s, dict[:base_ring])
     symbols = load_type_dispatch(s, Vector{Symbol}, dict[:symbols])
 
     if T <: PolyRing
         return polynomial_ring(base_ring, symbols..., cached=false)[1]
+    elseif T <: UniversalPolyRing
+        poly_ring = UniversalPolynomialRing(base_ring, cached=false)
+        gens(poly_ring, symbols)
+        return poly_ring
     end
 
     return polynomial_ring(base_ring, symbols, cached=false)[1]
 end
 
 ################################################################################
-# Multivariate Polynomials
+# Multivariate and Universal Polynomials
 @registerSerializationType(QQMPolyRingElem)
 @registerSerializationType(ZZMPolyRingElem)
```

Comment:
This is *not a review of this PR* but rather a separate question for @antonydellavecchia : I am actually surprised that we list `QQMPolyRingElem`, `ZZMPolyRingElem`, etc. separately here. Shouldn't these all be serialized as `MPolyRingElem`, with the `parent` and or `base_ring` determining the full type?

Reply:
it does seem like these are no longer needed, I'll make a seperate pull requests and get rid of some unnecessary registration